backend/persistent/plotly-api.js

```diff
@@ -164,36 +165,92 @@ export function patchGrid(fid, requestor, body) {
     });
 }
 
-export function updateGrid(rows, fid, uids, requestor) {
-    const {username, apiKey, accessToken} = getCredentials(requestor);
-
-    // TODO - Test case where no rows are returned.
-    if (uids.length !== (rows[0] || []).length) {
-        Logger.log(`
-            A different number of columns was returned in the
-            query than what was initially saved in the grid.
-            ${rows[0].length} columns were queried,
-            ${uids.length} columns were originally saved.
-            The connector does not create columns (yet),
-            and so we will only update the first ${uids.length}
-            columns.
-        `);
-    }
+export function updateGrid(rows, columnnames, fid, requestor) {
+    const numColumns = columnnames.length;
+    const columns = getColumns(rows, numColumns);
+    const columnEntries = columns.map((column, columnIndex) => {
+        return { data: column, name: columnnames[columnIndex], order: columnIndex };
+    });
 
-    /*
-     * For now, only update up to the number of columns that are
-     * already saved. In the future, we should just create more
-     * columns. See error message above.
-     */
-    const columns = getColumns(rows, uids.length);
-
-    const url = `grids/${fid}/col?uid=${uids.join(',')}`;
-    const body = {
-        cols: JSON.stringify(columns.map(column => ({
-            data: column
-        })))
-    };
-    return plotlyAPIRequest(url, {body, username, apiKey, accessToken, method: 'PUT'});
+    const {username, apiKey, accessToken} = getCredentials(requestor);
+    const baseUrl = `grids/${fid}/col`;
+    const baseParams = { username, apiKey, accessToken };
+
+    // fetch latest grid to get the source of truth
+    return getGrid(fid, requestor)
+        .then(res => {
+            if (res.status !== 200) {
+                return res;
+            }
+            return res.json();
+        }).then(data => {
+            if (data.status) {
```

Comment:
Elsewhere we throw an error, why not here?

Reply:
@mfix22 Judging by how we use `updateGrid` [here](https://github.com/plotly/falcon-sql-client/blob/9e5badbb502e88295bd1fac60da2929e5dce3599/backend/persistent/QueryScheduler.js#L295-L384), we shouldn't throw.